app/assets/javascripts/components/revisions/revisions_handler.jsx

```diff
@@ -8,17 +8,29 @@ const RevisionHandler = React.createClass({
 
   propTypes: {
     course_id: React.PropTypes.string,
-    course: React.PropTypes.object
+    course: React.PropTypes.object,
+  },
+
+  getInitialState() {
+    return {
+      limit: 50
+    };
   },
 
   componentWillMount() {
-    return ServerActions.fetch('revisions', this.props.course_id);
+    this.setState({ limit: 100 });
```

Comment:
This shouldn't be in componentWillMount, I think.

Reply:
Do you mean that it's a mistake or that shouldn't be placed there?

Reply:
This hook runs immediately when the component starts to load, so it would overwrite the default of 50 from getInitialState. I think you also can't set state from this React lifecycle hook anyway.

Reply:
I thought so but it works in my machine. I tried it with lower limit numbers and worked perfectly. componentWillMount was both taking the limit from the getInitialState and setting it on 100 for the next interaction. The same with the setState in showMore(), it worked as expected.

Reply:
I see we have setState a few places elsewhere within componentWillMount, so that seems to be okay technically. But it should be necessary as far as I can tell.